Netcode UC Project/Assets/App/Resource/Scripts/Player/ServerPlayerMovement.cs

```diff
@@ -39,15 +43,20 @@ void Update()
         {
             Move(moveInput, isPunching, isSprinting, isJumping);
         }
-        else if (IsClient)
+        else if (IsClient && !IsHost)
         {
             MoveServerRPC(moveInput, isPunching, isSprinting, isJumping);
         }
+
+        if(isPunching)
+        {
+            _bulletSpawner.FireProjectileRPC();
+        }
     }
 
     private void Move(Vector2 input, bool isPunching, bool isSprinting, bool isJumping)
     {
-        Vector3 moveDirection = input.x * _pTransform.right + input.y * _pTransform.forward;
+        moveDirection = new Vector3(input.x, 0f, input.y);
 
         _myAnimator.SetBool("IsWalking", moveDirection.x != 0 || moveDirection.y != 0);
 
```

Comment:
Plug this into your code right before the `if(isSprinting)` and just below the animation triggers and setbool. 
` if (input.x == 0f & input.y == 0f) return;`  this will allow you to hold and maintain your current position.